src/in-depth/render-configuration.md

```diff
@@ -0,0 +1,46 @@
+# Rendering documentation for your Configuration
+
+If your application offers extensive configuration options, you may want to
+render these for the conveniance of the user.
+
+You could render them straight to the terminal, so that a user can "ask" your
+app what configuration it provides:
+
+```commandline
+myapp print-config-options
+```
+
+Using the [type_description crate](https://crates.io/crates/type_description)
+you can annotate your configuration struct(s) and let Rust do the magic:
+
+```rust,ignore
+use type_description::AsTypeDescription;
+use type_description::TypeDescription;
+use type_description::TypeKind;
+use type_description::Sign;
+
+/// A configuration
+#[derive(TypeDescription)]
+struct Config {
+    /// The bind address
+    addr: std::net::SocketAddr,
+
+    /// The Port
+    port: u16,
+}
+```
+
```

Comment:
All code samples should be extracted from full programs that we compile in CI and can be run.